src/main/kotlin/org/rust/ide/inspections/fixes/ElideLifetimesFix.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.fixes
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.openapi.project.Project
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.childrenOfType
+import org.rust.lang.core.psi.ext.isRef
+import org.rust.lang.core.psi.ext.mutability
+
+class ElideLifetimesFix : LocalQuickFix {
+    override fun getName() = "Elide lifetimes"
+    override fun getFamilyName() = name
+
+    override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+        val fn = descriptor.psiElement as? RsFunction ?: return
+        LifetimeRemover().visitFunction(fn)
+    }
+}
+
+private class LifetimeRemover : RsVisitor() {
+    private val boundsLifetimes = mutableListOf<RsLifetimeParameter>()
+
+    override fun visitFunction(o: RsFunction) {
```

Comment:
IMO it's better to rename all `o` inside `visit` methods to more understandable names.